perun-wui-profile/src/main/java/cz/metacentrum/perun/wui/profile/pages/settings/altPasswords/AltPasswordsView.ui.xml

```diff
@@ -0,0 +1,74 @@
+             xmlns:p='urn:import:cz.metacentrum.perun.wui.widgets'
+             xmlns:g='urn:import:com.google.gwt.user.client.ui'
+             xmlns:b.gwt="urn:import:org.gwtbootstrap3.client.ui.gwt"
+>
+
+    <ui:style>
+    </ui:style>
+
+    <!-- Import CSS / GSS -->
+    <ui:with field="res" type="cz.metacentrum.perun.wui.profile.client.resources.PerunProfileResources"/>
+
+    <!-- Import Translation -->
+    <ui:with field='trans' type='cz.metacentrum.perun.wui.profile.client.resources.PerunProfileTranslation'/>
+
+    <!-- Import place tokens -->
+    <ui:with field="placeTokens" type="cz.metacentrum.perun.wui.profile.client.resources.PerunProfilePlaceTokens"/>
+
+    <b.html:Div addStyleNames="{res.gss.page}">
+
+        <b:Heading size="H2" marginTop="5">
+
+            <b:Breadcrumbs addStyleNames="{res.gss.perunBreadcrumb}">
+                <b:AnchorListItem icon="GEAR" text="{trans.menuSettings}"
+                                  targetHistoryToken="{placeTokens.getSettings}"/>
+                <b:ListItem text="{trans.menuAltPasswords}"/>
+            </b:Breadcrumbs>
+        </b:Heading>
+
+        <b.html:Div>
+            <p:PerunButton ui:field="backButton" text="{trans.back}" icon="CHEVRON_LEFT" type="DEFAULT"
+                           marginBottom="10" marginTop="10"/>
+        </b.html:Div>
+        <b:Alert ui:field="invalidValueAlert" type="DANGER" visible="false"/>
+        <b:Heading size="H4" text="{trans.generatePassword}"/>
+        <g:HorizontalPanel>
+            <b:TextBox ui:field="passwordBox" placeholder="{trans.passwordDescription}" />
+            <p:PerunButton ui:field="addAltPasswordButton" type="SUCCESS" text="{trans.addAltPassword}" icon="PLUS" block="true" marginLeft="10"/>
+        </g:HorizontalPanel>
+        <b:Heading size="H5" text="{trans.alternativePasswordsExample}" marginBottom="20"/>
+
+        <b:Heading ui:field="alternativePasswordsInfo" text="{trans.yourAlternativePasswords}" size="H4"/>
+        <g:ResizeLayoutPanel addStyleNames="{res.gss.grid}">
+            <b.gwt:DataGrid bordered="true" condensed="true" striped="true" hover="true" ui:field="aPDataGrid"/>
+        </g:ResizeLayoutPanel>
+
+        <b.html:Div ui:field="panelsDiv">
+
+            <b:Modal ui:field="generatedPasswordModal" closable="false" fade="true"
+                     dataBackdrop="STATIC" dataKeyboard="true">
+                <b:ModalHeader>
+                    <b:Heading size="H3" text="{trans.generatedPassword}" alignment="CENTER"/>
+                </b:ModalHeader>
+                <b:ModalBody>
+                    <b:Alert ui:field="generatedPasswordAlert" type="INFO" visible="false"/>
+                    <b:Heading size="H5" text="{trans.generatedPasswordHint}" alignment="CENTER"/>
+                    <b:TextBox ui:field="generatedPasswordBox" readOnly="true"/>
+                </b:ModalBody>
+                <b:ModalFooter>
+                    <b:ButtonGroup pull="RIGHT">
```

Comment:
Please do not group buttons which perform different actions. You can wrap each in own ButtonGroup and also you can wrap it all in ``<b:ButtonToolBar>``. Also copy to clipboard button should have positive color, since its preferred action, so either SUCCESS type or PRIMARY.